usaspending_api/agency/v2/views/awards.py

```diff
@@ -0,0 +1,81 @@
+from django.utils.functional import cached_property
+from elasticsearch_dsl import A
+from rest_framework.response import Response
+from usaspending_api.agency.v2.views.agency_base import AgencyBase
+from usaspending_api.awards.v2.lookups.lookups import award_type_mapping
+from usaspending_api.common.cache_decorator import cache_response
+from usaspending_api.common.elasticsearch.search_wrappers import TransactionSearch
+from usaspending_api.common.query_with_filters import QueryWithFilters
+
+
+class Awards(AgencyBase):
+    """
+    Returns award and transaction information for USAspending.gov's Agency Details page for agencies
+    that have ever awarded.
+    """
+
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/agency/toptier_code/awards.md"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.additional_models = [
+            {
+                "name": "award_type",
+                "key": "award_type",
+                "type": "array",
+                "array_type": "enum",
+                "enum_values": list(award_type_mapping.keys()) + ["no intersection"],
+                "optional": True,
+            },
+            {
+                "name": "agency_type",
+                "key": "agency_type",
+                "type": "enum",
+                "enum_values": ("awarding", "funding"),
+                "default": "awarding",
+            },
+        ]
+        self.params_to_validate = ["fiscal_year", "award_type", "agency_type"]
+
+    @cached_property
+    def _query_params(self):
```

Comment:
Talked about in Slack but capturing here.

Move this into the AgencyBase since it will be used by a few different endpoints.